client/app/email-domain/email/account/migrate/email-domain-email-account-migrate.controller.js

```diff
@@ -0,0 +1,152 @@
+                    }
+
+                    this.loading = false;
+                });
+        }
+
+        // Request services list from service type
+        getDestinationServicesFromType (type) {
+            return this.Emails.getDestinationServices(this.email.domain, this.email.accountName, type)
+                .catch((err) => this.handleError(err));
+        }
+
+        // Get Destination email (@configure.me)
+        getDestinationEmails (destinationService) {
+            // Check for destination service
+            if (angular.isUndefined(destinationService)) {
+                this.loading = false;
+                return;
+            }
+
+            this.loading = true;
+
+            // Reset for ngChange
+            this.destinationEmails = [];
+            this.remainingAvailableEmails = null;
+            this.migrate.destinationEmail = null;
+
+            // Get available email addresses for the migration
+            this.Emails.getDestinationEmailAddresses(this.email.domain, this.email.accountName, destinationService)
+                .then((data) => {
+                    this.loading = false;
+                    this.destinationEmails = data;
+
+                    // Auto select the first email
+                    if (this.destinationEmails && this.destinationEmails.length > 0) {
+                        this.remainingAvailableEmails = this.destinationEmails.length - 1;
+                        this.migrate.destinationEmail = this.destinationEmails[0];
+                    }
+                })
+                .catch((err) => this.handleError(err))
+                .finally(() => { this.loading = false; });
+        }
+
+        // Check if it's possible to migrate
+        checkMigrationData () {
+            this.loading = true;
+
+            this.Emails.checkMigrate(this.email.domain, this.email.accountName, this.migrate.destinationService, this.migrate.destinationEmail)
+                .then(() => { this.loading = false; })
+                .catch((err) => this.handleError(err));
+        }
+
+        // Post request for migration
+        migrateAccount () {
+            this.Emails.migrateAccountToDestinationAccount(this.email.domain, this.email.accountName, this.migrate.destinationService, this.migrate.destinationEmail, this.migrate.password)
+                .then(() => this.Alerter.success(this.$scope.tr("email_tab_modal_migrate_success"), this.$scope.alerts.dashboard))
+                .catch((err) => this.handleError(err))
+                .finally(() => this.$scope.resetAction());
+        }
+
+        // Check from email-domain-email-account-change-password.controller.js
+        checkPassword (input) {
+            input.$setValidity("passwordCheck", !!this.migrate.password && !/^\s/.test(this.migrate.password) && !/\s$/.test(this.migrate.password) && !this.migrate.password.match(/[ÂÃÄÀÁÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝÞßàáâãäåæçèéêëìíîïðñòóôõöøùúûüýþÿ]/));
```

Comment:
Does `ng-messages` & `ng-pattern` can help you in the same way?

Reply:
Already seen with Joffrey. It's a copypaste from the modal "Change password"